src/app/static/src/app/components/ResetConfirmation.vue

```diff
@@ -1,36 +1,58 @@
 <template>
-    <modal :open="open">
-        <h4 v-translate="'haveYouSaved'"></h4>
-        <p v-translate="'discardWarning'"></p>
-        <ul>
-            <li v-for="step in laterCompleteSteps">
-                Step {{step.number}}: <span v-translate="step.textKey"></span>
-            </li>
-        </ul>
-        <p v-translate="'savePrompt'"></p>
-        <template v-slot:footer>
-            <button type="button"
-                    class="btn btn-white"
-                    @click="continueEditing"
-                    v-translate="'discardSteps'">
-            </button>
-            <button type="button"
-                    class="btn btn-red"
-                    @click="cancelEditing"
-                    v-translate="'cancelEdit'">
-            </button>
-        </template>
-    </modal>
+    <div>
+        <modal :open="open">
+            <h4 v-if="guestUser" v-translate="'haveYouSaved'"></h4>
+            <h4 v-if="!guestUser">Save snapshot?</h4>
+
+            <p v-translate="'discardWarning'"></p>
+            <ul>
+                <li v-for="step in laterCompleteSteps">
+                    Step {{step.number}}: <span v-translate="step.textKey"></span>
+                </li>
+            </ul>
+
+            <p v-if="guestUser"  v-translate="'savePrompt'"></p>
+            <p v-if="!guestUser">
+                These steps will automatically be saved in a snapshot. You will be able to reload this snapshot from the Versions page.
+            </p>
+
+            <template v-if="!waitingForSnapshot" v-slot:footer>
+                <button type="button"
+                        class="btn btn-white"
+                        @click="handleConfirm"
+                        v-translate="guestUser? 'discardSteps' : 'saveSnapshotConfirm'">
+                </button>
+                <button type="button"
+                        class="btn btn-red"
+                        @click="cancelEditing"
+                        v-translate="guestUser? 'cancelEdit': 'cancelEditLoggedIn'">
+                </button>
+            </template>
+
+            <div v-if="waitingForSnapshot" class="text-center">
+                <loading-spinner size="sm"></loading-spinner>
+                <h4 id="spinner-text">Saving snapshot</h4>
+            </div>
+        </modal>
+    </div>
 </template>
 
 <script lang="ts">
     import Vue from "vue";
     import Modal from "./Modal.vue";
-    import {mapGetterByName} from "../utils";
+    import {mapActionByName, mapGetterByName, mapStateProp} from "../utils";
     import {StepDescription} from "../store/stepper/stepper";
+    import LoadingSpinner from "./LoadingSpinner.vue";
+    import {VersionsState} from "../store/versions/versions";
+    import {ErrorsState} from "../store/errors/errors";
+
+    declare const currentUser: string;
```

Comment:
I feel like this hack is in enough places now that we should sort out getting it into the store state - have written a ticket https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1743

Reply:
Good call